contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -116,34 +123,30 @@ contract ItemsRolesRegistryFacet is Modifiers, ISftRolesRegistry, ERC1155Holder
         }
 
         _unequipAllDelegatedWearables(_nonce, _grantor, _depositInfo.tokenId);
-        
-        s.itemsDepositsUnequippedBalance[_grantor][_nonce] = s.itemsDeposits[_grantor][_nonce].tokenAmount;
-        delete s.itemsRoleAssignments[_grantor][_nonce];
 
-        emit RoleRevoked(
-            _nonce,
-            UNIQUE_ROLE,
-            _depositInfo.tokenAddress,
-            _depositInfo.tokenId,
-            _depositInfo.tokenAmount,
-            _grantor,
-            _roleData.grantee
-        );
+        _userInfo.availableBalance = _userInfo.deposit.tokenAmount;
+        delete _userInfo.roleAssignment;
+
+        emit RoleRevoked(_nonce, UNIQUE_ROLE, _depositInfo.tokenAddress, _depositInfo.tokenId, _depositInfo.tokenAmount, _grantor, _roleData.grantee);
     }
 
-    function withdrawFrom(uint256 _nonce, address _grantor) override external onlyOwnerOrApproved(_grantor, s.itemsDeposits[_grantor][_nonce].tokenAddress) {
-        DepositInfo memory _depositInfo = s.itemsDeposits[_grantor][_nonce];
+    function withdrawFrom(
+        uint256 _nonce,
+        address _grantor
+    ) external override onlyOwnerOrApproved(_grantor, s.userDelegatedItemsInfo[_grantor][_nonce].deposit.tokenAddress) {
+        UserDelegatedItemsInfo storage _userInfo = s.userDelegatedItemsInfo[_grantor][_nonce];
+        DepositInfo memory _depositInfo = _userInfo.deposit;
         require(_depositInfo.tokenAmount > 0, "ItemsRolesRegistryFacet: nonce does not exist");
         require(
-            s.itemsRoleAssignments[_grantor][_nonce].expirationDate < block.timestamp || s.itemsRoleAssignments[_grantor][_nonce].revocable,
+            _userInfo.roleAssignment.expirationDate < block.timestamp || _userInfo.roleAssignment.revocable,
             "ItemsRolesRegistryFacet: token has an active role"
         );
-        
+
         _unequipAllDelegatedWearables(_nonce, _grantor, _depositInfo.tokenId); // If the item is equipped in some gotchi, it will be unequipped
-        
-        delete s.itemsDeposits[_grantor][_nonce];
-        delete s.itemsDepositsUnequippedBalance[_grantor][_nonce];
-        delete s.itemsRoleAssignments[_grantor][_nonce];
+
+        delete _userInfo.deposit;
+        delete _userInfo.availableBalance;
+        delete _userInfo.roleAssignment;
```

Comment:
Simply delete `_userInfo`

Reply:
I just checked and we cannot delete a struct of structs 

<img width="788" alt="Screenshot 2023-12-20 at 12 43 44" src="https://github.com/OriumNetwork/aavegotchi-contracts/assets/38267570/0602eac0-2fa5-420a-971a-1b10eea29458">